sphinx/search/__init__.py

```diff
@@ -350,9 +353,10 @@ def prune(self, filenames):
         for wordnames in itervalues(self._title_mapping):
             wordnames.intersection_update(filenames)
 
-    def feed(self, filename, title, doctree):
+    def feed(self, docname, filename, title, doctree):
```

Comment:
I'd read the error logs on Travis CI roughly. It seems some `TypeError`s come from this new argument `docname`.
```
...
  File "/home/travis/build/sphinx-doc/sphinx/tests/test_search.py", line 42, in test_wordcollector
    ix.feed('filename', 'title', doc)
TypeError: feed() takes exactly 5 arguments (4 given)
```
```
...
  File "/home/travis/build/sphinx-doc/sphinx/sphinx/builders/websupport.py", line 79, in write_doc_serialized
    self.index_page(docname, doctree, title)
  File "/home/travis/build/sphinx-doc/sphinx/sphinx/builders/html.py", line 730, in index_page
    self.indexer.feed(pagename, filename, title, doctree)
TypeError: feed() takes exactly 4 arguments (5 given)
```

And it might have caused the more errors on websupport module.
Could you check this first?